### PR TITLE
[auth] Make email unique on auth service

### LINKED ISSUE #26: Duplicate users
Users are allowed to sign up with the same email address multiple times to the `/auth` POST endpoint. We should probably not allow this...

### PR DESCRIPTION
**Pull request by jaylees14**
Closes #26

### CODE REVIEW

File: auth/dao/dao.go [hunk 1]
```diff
@@ -78,14 +82,16 @@ func executeQueryWithRowResponse(db *sql.DB, query string, args ...interface{})
...
+	err := row.Scan(&auth.ID, &auth.Email, &auth.Password)
 	if err != nil {
-		switch err {
-		case sql.ErrNoRows:
-			return nil, ErrAuthNotFound
-		default:
-			return nil, err
+		// PQ specific error
+		if err, ok := err.(*pq.Error); ok {
+			log.Printf("%s", err.Code)
```
Comment (c-harding) [inline auth/dao/dao.go]: DELET THIS PLZ
```suggestion
```

**Review by c-harding — APPROVED** [review]
LGTM \ suggestions

File: auth/dao/dao.go [hunk 2]
```diff
@@ -86,7 +85,7 @@ func (dao *DAO) CreateAuth(request AuthCreateRequest) (*Auth, error) {
 	if err != nil {
 		// PQ specific error
 		if err, ok := err.(*pq.Error); ok {
-			if err.Code == psqlDuplicateKey {
+			if err.Code.Name() == psqlUniqueViolation {
```
Comment (c-harding) [inline auth/dao/dao.go:88]: Much nicer

**Review by c-harding — APPROVED** [review]

File: auth/dao/dao.go [hunk 3]
```diff
@@ -78,14 +81,15 @@ func executeQueryWithRowResponse(db *sql.DB, query string, args ...interface{})
...
-	err := row.Scan(&auth.Id, &auth.Email, &auth.Password)
+	err := row.Scan(&auth.ID, &auth.Email, &auth.Password)
 	if err != nil {
-		switch err {
-		case sql.ErrNoRows:
-			return nil, ErrAuthNotFound
-		default:
-			return nil, err
+		// PQ specific error
+		if err, ok := err.(*pq.Error); ok {
```
Comment (Lewiky) [inline auth/dao/dao.go:87]: This is awful

**Review by Lewiky — APPROVED** [review]

**Pull request merged**